src/SeleniumLibrary/base/context.py

```diff
@@ -53,13 +55,22 @@ def find_element(self, locator, tag=None, required=True, parent=None):
         :param parent: Optional parent `WebElememt` to search child elements
             from. By default search starts from the root using `WebDriver`.
         :type parent: selenium.webdriver.remote.webelement.WebElement
+        :param warn: If true, will log at warn level if locator did match to
+            multiple WebElement's. Used to give information to users if locator
+            did match to multiple elements.
+        :type warn: True or False
         :return: Found `WebElement` or `None` if element not found and
             `required` is false.
         :rtype: selenium.webdriver.remote.webelement.WebElement
         :raises SeleniumLibrary.errors.ElementNotFound: If element not found
             and `required` is true.
         """
-        return self.element_finder.find(locator, tag, True, required, parent)
+        element, multiple = self.element_finder.find(locator, tag, True,
+                                                     required, parent)
+        if multiple and warn:
```

Comment:
so if I am selecting the collection of li which have class "foobar" a selector "css=.foobar"  this will warn?

Reply:
That is the basic idea, but we should not log the warning always. The full idea is, if keyword should find single element, then it should display warning that multiple elements where found. Because the element the SeleniumLibrary will choose, it always will choose first one returned by Selenium, may not be the correct one to choose. This is meant to protect the users against changes in the DOM and for the users new to the Selenium automation. Example for `Click Button` there would be warning logged, but not for `Page Should Contain`. 

The PR is not yet ready, because now too many keywords will log the warning and I would have to choose which keywords actually will log the warning. But I am seeking for a input, exactly like this, that am I going to the right direction for implementing the feature. Also I am looking technical input, because there is many ways to implement the feature and I have changed my mind few times, where I should I implement the feature. 

When I did write #978, the feature idea did sounded like a great idea for me. Now when implementing the feature, I am having second thoughts. But those are mainly because I can not decide where to the technical implementation should go. In any case, I would love to get the feedback.